private/bufpkg/bufapimodule/module_reader.go

```diff
@@ -56,7 +70,7 @@ func (m *moduleReader) GetModule(ctx context.Context, modulePin bufmoduleref.Mod
 		moduleIdentity,
 		modulePin.Commit(),
 	)
-	if resp.Manifest != nil {
+	if m.tamperProofingEnabled && resp.Manifest != nil {
```

Comment:
I think we should fail here and not fallback:

```go
if m.tamperProofingEnabled {
  if resp.Manifest == nil {
    // not fallback to proto module
    return nil, errors.New("we're expecting manifest+blobs, BSR didn't send them")
  }
  // process manifest+blobs
  return ...
}

// flag is off
// should we make sure the manifest+blobs were not sent? or ignore them?
// deal with resp.Module
return...
```

Reply:
Good call - updated.